ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,292 @@
+    }
+    $row = array(
+      array(
+        'data' => check_plain($record->payment_order_id),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_transaction_id),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_time),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->customer_uid),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($text),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => check_plain($record->payment_price),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->payment_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->provider_payment_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => $record->dibs_capture_status ? t('yes') : t('no'),
+        'class' => 'payment',
+      ),
+      array(
+        'data' => '',
+        'class' => 'icon ' . $status,
+      ),
+    );
+    $rows[] = $row;
+    $no_transactions_msg = FALSE;
+  }
+
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+  ));
+  if ($no_transactions_msg) {
+    $output .= $no_transactions_msg;
+  }
+  return render($output);
+}
+
+/**
+ * Callback that builds page reports run cron.
+ *
+ * @return
+ *   Returns tabular representation of values results.
+ */
+function ding_dibs_cron_report() {
+  drupal_set_title(t('Ding DIBS cron report'));
+  $report = db_select('ding_dibs_report_cron', 'd')
+    ->fields('d', array('id', 'time_start', 'status', 'text'))
+    ->orderBy('id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  $no_transactions_msg = '<p>' . t('No cron report.') . '</p>';
+  if (!$max_count) {
+    return $no_transactions_msg;
+  }
+  $rows = array();
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $header = array(array('data' => t('Message'), 'class' => 'payment'));
+  foreach ($report as $record) {
+    if (empty($record->text)) {
+      $day = format_date($record->time_start, 'custom', 'd');
+      $hour = format_date($record->time_start, 'custom', 'H:i');
+      $record->text = $day . 'th september ' . $hour . ': Job failed.';
+    }
+    $row = array(
+      array(
+        'data' => t($record->text),
+        'class' => 'payment',
+      ),
+    );
+    $rows[] = $row;
+    $no_transactions_msg = NULL;
+  }
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+  ));
+  if ($no_transactions_msg) {
+    $output .= $no_transactions_msg;
+  }
+  return render($output);
+}
+
+/**
+ * Generate an array of leading 0.
+ *
+ * @param $item
+ *   Input array.
+ *
+ * @return
+ *   Returns an array with the format of the time 01, 02.
+ */
+function generate_zerro ($item, $key) {
```

Comment:
There are some issues with this function:

- Functions in modules should be prefixed with the module name e.g. `_ding_dibs_generate_zero()`.
- The argument `$key` is declared but not documented and never used.
- The DocBlock makes little sense compared to the implementation. There is neither array input nor output.